File: src/main/java/io/bindernews/thegrackle/cards/FiredUpCard.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import io.bindernews.bnsts.CardNums;
import io.bindernews.thegrackle.power.FiredUpPower;

public class FiredUpCard extends BaseCard {
    public static final CardConfig C = new CardConfig("FiredUpCard", CardType.POWER);
    public static final CardNums NUM = CardNums.builder()
            .cost(2).costUpg(1).magic(8).build();

    public FiredUpCard() {
        super(C, CardRarity.RARE, CardTarget.SELF);
        NUM.init(this);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        addToBot(new ApplyPowerAction(p, p, new FiredUpPower(p, magicNumber)));
    }

    @Override
    public void upgrade() {
        NUM.upgrade(this, false);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/FiredUpCard.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.power.FiredUpPower

class FiredUpCard : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        addToBot(ApplyPowerAction(p, p, FiredUpPower(p, magicNumber)))
    }

    companion object {
        @JvmField val C = CardConfig("FiredUpCard", CardType.POWER, CardRarity.RARE, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(2, 1)
            c.magic(8, -1)
        }
    }
}